src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -210,16 +212,27 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
 
                         module.ModuleName = new string(chars, 0, length);
 
-                        length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                        for (; ; )
+                        {
+                            length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                            if (length == chars.Length)
+                            {
+                                ArrayPool<char>.Shared.Return(chars);
+                                minimumLength *= 2;
+                                chars = ArrayPool<char>.Shared.Rent(minimumLength);
```

Comment:
This is a problematic pattern.  You could end up returning chars and then if the Rent throws an exception, the finally block will return chars again, resulting in the same array getting returned to the pool twice.